docs/synapse/userguides/storm_adv_functions.rstorm

```diff
@@ -0,0 +1,708 @@
+  }
+
+
+Example
+*******
+
+Some data sources may paginate results, returning X number of objects (e.g., in a JSON blob) at a time
+until all results are returned. A data emitter function can emit individual JSON objects from the blob
+(e.g., for consumption by another function that processes the object and creates nodes) until all of
+the results have been received.
+
+::
+  
+  function emitReportFeed() {
+      
+      //Set variables for the current time and the # of objects to retrieve per page
+      $now = $lib.time.now()
+      $pagelim = 100
+      
+      //Set a variable for API query parameters
+      $params = ({
+          "limit": $pagelim,
+      })
+      
+      //Set a variable for the initial offset
+      $offset = (0)
+      
+      //While loop to retrieve records
+      while (1) {
+          
+          //Set the value of the 'offset' parameter
+          $params.offset = $offset
+          
+          //Invoke an existing function to retrieve the JSON using $pararms as parameters
+          // to the API request.
+          //Assign the returned JSON to the variable $data
+          $data = $getJson("/reports", params=$params)
+          
+          //If no data is returned, stop and exit this function
+          if ($data = $lib.null) { stop }
+          
+          //If data is returned, loop over the JSON and emit each item / ojbect
+          for $item in $data.data { emit $item }
+          
+          //Set $datasize to the size (number of items) in the returned JSON
+          $datasize = $data.data.size()
+          
+          //Check whether the # of records returned is less than our page limit
+          //If so we have retrieved all available records
+          if ($datasize < $pagelim) {
+                        
+              //Print status to stdout if debug is in use
+              if $lib.debug { $lib.print(`Reports ingested up to {$now}`) }
+              
+              //Invoke an existing function to update the 'last retrieved' date to the current time
+              //E.g., this value may be stored as node data on the feed's meta:source node
+              $setReportFeedLast($now)
+              
+              //Stop and exit the function
+              stop
+          }
+          
+          //If $datasize is NOT < $pagelim there is more data
+          //Update the $offset value and execute the while loop again
+          $offset = ($offset + $pagelim)
+     }
+   }
+
+.. _storm-func-yielder:
+
+Node Yielder Functions
+++++++++++++++++++++++
+
+**Node yielder** functions yield nodes. If a function **does not** include either of the keywords ``return``
+or ``emit``, it is presumed to be a node yielder.
+
+Node yielder functions **stream** nodes; (technically, they return a **generator object** that is iterated
+over). They are designed to yield nodes as they are available while continuing to execute. They may be
+invoked with **for** loops for this purpose.
```

Comment:
You can also just `yield` a node yielder function to yield the nodes into the pipeline, doesn't have to be a "for" loop

Reply:
I guess yielding into the pipeline is covered elsewhere so maybe it isn't necessary here as well.

Reply:
revised wording